### DISCUSSION TITLE
Torchscript backward compatibility

### DISCUSSION THREAD

**Discussion by gcuendet** [pytorch/TensorRT#1819] [Q&A] [answered]
Hi! 

What is your policy regarding torchscript backward compatibility?

What I mean by that is: is it supported to export a torchscript using an "old" version of pytorch, and then try to convert that torchscript using torch-tensorRT newest version? As a practical example, I see different behaviours when trying to convert a torchscript exported using pytorch 1.9 or pytorch 1.11 with torch-tensorRT 1.3. It's failing with both versions of the torchscript:
- with [an error](https://github.com/pytorch/TensorRT/issues/1687) regarding upsample_bilinear2d when using the torchscript generated with pytorch 1.9
- not failing with an error but hanging forever at the Graph Lowering step when using the torchscript generated with pytorch 1.11.

Thanks in advance for any feedback you might have.

**Comment by narendasan** [comment, marked as answer]
Generally there is not backwards compatibility between versions of PyTorch. This is due to PyTorch in the past changing their IR from version to version. Therefore, we always recommend exporting your torchscript with the version you intend to deploy with.
